edml-1.0.0.json

```diff
@@ -0,0 +1,215 @@
+      "minProperties": 1,
+      "additionalProperties": {
+        "$ref": "#/definitions/mappingDefinition"
+      }
+    },
+    "destinationTable": {
+      "type": "string",
+      "title": "Name of the destination table in Exasol Virtual Schema",
+      "description": "The name of the resulting table in Exasol.",
+      "pattern": "^[a-zA-Z0-9_]+$"
+    },
+    "mappingDefinition": {
+      "type": "object",
+      "description": "A mapping definition can either be:\n - a mapping to a column or table\n - an step  into the property tree of the DynamoDB table",
+      "minProperties": 1,
+      "maxProperties": 1,
+      "properties": {
+        "fields": {
+          "$ref": "#/definitions/fields"
+        },
+        "toVarcharMapping": {
+          "type": "object",
+          "description": "Maps the selected DynamoDB property to an Exasol VARCHAR column.\n\n*Supported DynamoDB types:* String (S), Number (N), Boolean (BOOL)",
+          "properties": {
+            "varcharColumnSize": {
+              "$ref": "#/definitions/varcharColumnSize"
+            },
+            "overflowBehaviour": {
+              "default": "TRUNCATE",
+              "description": "Behaviour to apply in case a value exceeds the size of the VARCHAR column.\n\nTRUNCATE: truncate the string NULL: use null instead ABORT: abort whole query",
+              "enum": [
+                "TRUNCATE",
+                "ABORT",
+                "NULL"
+              ]
+            },
+            "destinationName": {
+              "type": "string",
+              "description": "Name of the Exasol VARCHAR column"
+            },
+            "description": {
+              "$ref": "#/definitions/description"
+            },
+            "required": {
+              "default": false,
+              "$ref": "#/definitions/required"
+            },
+            "key": {
+              "$ref": "#/definitions/key"
+            }
+          },
+          "additionalProperties": false
+        },
+        "toJsonMapping": {
+          "title": "toJSON mapping definition",
+          "description": "Maps the selected DynamoDB property and all its descendants to a JSON string.\n\n*Supported DynamoDB types:* Map (M), List (L). The root document is also supported.",
+          "properties": {
+            "varcharColumnSize": {
+              "$ref": "#/definitions/varcharColumnSize"
+            },
+            "overflowBehaviour": {
+              "description": "Behaviour to apply in case a value exceeds the size of the Exasol column.\n\nNULL: use null instead ABORT: abort whole query",
+              "default": "ABORT",
+              "enum": [
+                "NULL",
+                "ABORT"
+              ]
+            },
+            "destinationName": {
+              "type": "string",
+              "description": "Name of the Exasol column. required if used at root level. Else the name of the DynamoDB property is used."
```

Comment:
In that case modelling the JSON schem with two variants depending on the position in the tree is cleaner.

Reply:
That would mean to duplicate the `toJsonMapping` and the `mappingDefinition` including references to all their children. I think that does not justify the small gain in UX.